code/go/0chain.net/smartcontract/stakepool/unlock.go

```diff
@@ -38,10 +50,29 @@ func (sp *StakePool) UnlockPool(
 	dpUpdate.Updates["status"] = spenum.Deleting
 	dpUpdate.emitUpdate(balances)
 
+	err = sp.EmitUnStakeEvent(providerType, providerId, amount, balances)
+	if err != nil {
+		return "", fmt.Errorf(
+			"stake pool staking error: %v", err)
+	}
 	balances.EmitEvent(event.TypeStats, event.TagUnlockStakePool, clientID, lock)
 	if err != nil {
 		return "", fmt.Errorf("error emptying account, %v", err)
 	}
 
 	return toJson(lock), nil
 }
+
+func (sp *StakePool) EmitUnStakeEvent(providerType spenum.Provider, providerID string, amount currency.Coin, balances cstate.StateContextI) error {
+	logging.Logger.Info("emitting stake event")
+
+	h, ok := unstakeHandlers[providerType]
+	if !ok {
+		logging.Logger.Error("unsupported providerType in stakepool StakeEvent")
+		return nil
```

Comment:
Return error or remove the result error as this function does not return any error actually.